courses/views_test.py

```diff
@@ -0,0 +1,188 @@
+"""
```

Comment:
These tests are very copy/paste-heavy, but that felt more appropriate than creating some abstraction. The view logic between these models will probably drift in the future, in which case it will help to have separate test cases for each.